src/Server/Services/Analytics/AnalyticsService.lua

```diff
@@ -622,31 +893,29 @@ function AnalyticsService:LogPurchase(data: PurchaseEvent): { [any]: any }
 		elseif data.flowType ~= nil and not GameAnalytics.EGAResourceFlowType[data.flowType] then
 			return reject("flow type is invalid")
 		end
-		
+
 		self:LogResourceEvent({
 			userId = data.userId,
 			amount = data.amount or 1,
 			currency = data.currency,
 			flowType = (
-				data.flowType == GameAnalytics.EGAResourceFlowType.Source and
-				GameAnalytics.EGAResourceFlowType.Source
-			) or (
-				data.flowType == GameAnalytics.EGAResourceFlowType.Sink and
-				GameAnalytics.EGAResourceFlowType.Sink
-			) or
-				GameAnalytics.EGAResourceFlowType.Sink,
+				data.flowType == GameAnalytics.EGAResourceFlowType.Source
+				and GameAnalytics.EGAResourceFlowType.Source
+			)
+				or (data.flowType == GameAnalytics.EGAResourceFlowType.Sink and GameAnalytics.EGAResourceFlowType.Sink)
+				or GameAnalytics.EGAResourceFlowType.Sink,
 			eventType = data.eventType,
-			itemId = data.itemId
+			itemId = data.itemId,
 		})
-		
-		return resolve()
+
+		resolve()
```

Comment:
```suggestion
		return resolve()
```
The linter/type checker complains if there's a return type defined for a function, but the promise is resolved/rejected without returning the function (For some reason)? It's an invalid error since we know we're returning a promise, but the code checker flags it for some reason.